### PR TITLE
WC-1032 Implemented agent tokens validation on connect to server.

### PR DESCRIPTION
**Pull request by akuznetsov-gridgain**

### CODE REVIEW

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java [hunk 1]
```diff
@@ -185,11 +200,40 @@ private void ping(WebSocketSession ws) {
...
+
+            Set<String> validTokens = accRepo.validateTokens(tokens);
 
-        log.info("Agent connected: " + agentInfo);
+            if (F.isEmpty(validTokens)) {
+                res.setError("Failed to authenticate with token(s): " + tokens + "." +
+                    " Please reload agent or check settings.");
+            }
+            else {
+                req.setTokens(validTokens);
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: Why you modify request object?

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java [hunk 2]
```diff
@@ -185,11 +200,40 @@ private void ping(WebSocketSession ws) {
...
+                res.setTokens(validTokens);
+            }
+        }
+
+        if (F.isEmpty(res.getError())) {
+            log.info("Agent connected: " + req);
+
+            agents.put(ws, req);
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: Why we save full request object if we use only tokens?

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java [hunk 3]
```diff
@@ -185,11 +200,40 @@ private void ping(WebSocketSession ws) {
      * @throws IOException If failed to process.
      */
     private void registerAgent(WebSocketEvent evt, WebSocketSession ws) throws IOException {
-        AgentInfo agentInfo = fromJson(evt.getPayload(), AgentInfo.class);
+        AgentHandshakeRequest req = fromJson(evt.getPayload(), AgentHandshakeRequest.class);
+
+        AgentHandshakeResponse res = new AgentHandshakeResponse();
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: I think will be better to create separate constructor or builder method for error and response.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java [hunk 4]
```diff
@@ -185,11 +200,40 @@ private void ping(WebSocketSession ws) {
      * @throws IOException If failed to process.
      */
     private void registerAgent(WebSocketEvent evt, WebSocketSession ws) throws IOException {
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: Why you pass WebSocketSession, WebSocketEvent? Code in this method should only validate and construct response object.

**Review by nva — CHANGES_REQUESTED** [review]

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java (hunk 1 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: Fixed

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java (hunk 2 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: Fixed.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java (hunk 3 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: Fixed.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java (hunk 4 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: This allows to make "handleAgentEvents()" method simple.
"registerAgent()" - is a private method, I like have deserialization of AgentHandshakeRequest in this method for now.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java [hunk 5]
```diff
@@ -180,16 +195,62 @@ private void ping(WebSocketSession ws) {
...
 
     /**
-     * @param evt Event to process.
      * @param ws Session.
+     * @param evt Event to process.
      * @throws IOException If failed to process.
      */
-    private void registerAgent(WebSocketEvent evt, WebSocketSession ws) throws IOException {
-        AgentInfo agentInfo = fromJson(evt.getPayload(), AgentInfo.class);
+    private void registerAgent(WebSocketSession ws, WebSocketEvent evt) throws IOException {
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: Why this method receive `WebSocketSession` and `WebSocketEvent`? You can remove nested `if-else` if this method only receive `AgentHandshakeRequest` and return `AgentHandshakeResponse`

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java (hunk 4 above)
Comment (nva) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: But now your business logic depends from socket implementation, have many nested `if-else`. Look at rest-http module we don't pass Servlets specific code into `CommandHandlers`

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java (hunk 5 above)
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java]: But now your business logic depends from socket implementation, have many nested if-else. Look at rest-http module we don't pass Servlets specific code into CommandHandlers

**Review by nva — CHANGES_REQUESTED** [review]

**Pull request closed without merging**